api_server/src/main/java/io/zetch/app/web/ReplyController.java

```diff
@@ -0,0 +1,104 @@
+package io.zetch.app.web;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.zetch.app.domain.reply.ReplyEntity;
+import io.zetch.app.domain.reply.ReplyGetDto;
+import io.zetch.app.domain.reply.ReplyPostDto;
+import io.zetch.app.service.ReplyService;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping(path = "/replies")
+@Tag(name = "Replies")
+@CrossOrigin(origins = "*") // NOSONAR
+public class ReplyController {
+  private final ReplyService replyService;
+  private final ObjectMapper mapper =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  @Autowired
+  public ReplyController(ReplyService replyService) {
+    this.replyService = replyService;
+  }
+
+  @PostMapping(path = "/")
+  @Operation(summary = "Create a new reply")
+  @SecurityRequirement(name = "OAuth2")
+  @ResponseBody
+  ReplyGetDto addNewReply(@RequestBody ReplyPostDto newReply) throws JsonProcessingException {
+    ReplyEntity reply =
+        replyService.createNew(
+            newReply.getReplyComment(), newReply.getReplyUserId(), newReply.getReviewId());
+    String serialized = mapper.writeValueAsString(reply);
```

Comment:
Defining toGetDto function like this would be bug preventing.
https://github.com/astrohsy/ZETCH/blob/aa56fa3c2f5b78dc1d91b62af905d9a94d8d7d19/api_server/src/main/java/io/zetch/app/domain/review/ReviewEntity.java#L48-L50

Then you can use that instance method to get DTO like below
https://github.com/astrohsy/ZETCH/blob/aa56fa3c2f5b78dc1d91b62af905d9a94d8d7d19/api_server/src/main/java/io/zetch/app/web/ReviewController.java#L50